OpenRA.Game/Server/Server.cs

```diff
@@ -129,15 +126,43 @@ public void EndGame()
 				t.GameEnded(this);
 		}
 
-		public Server(IPEndPoint endpoint, ServerSettings settings, ModData modData, ServerType type)
+		public Server(List<IPEndPoint> endpoints, ServerSettings settings, ModData modData, ServerType type)
 		{
 			Log.AddChannel("server", "server.log", true);
 
-			listener = new TcpListener(endpoint);
-			listener.Start();
-			var localEndpoint = (IPEndPoint)listener.LocalEndpoint;
-			Ip = localEndpoint.Address;
-			Port = localEndpoint.Port;
+			SocketException lastException = null;
+			var checkReadServer = new List<Socket>();
+			foreach (var endpoint in endpoints)
+			{
+				var listener = new TcpListener(endpoint);
+				try
```

Comment:
why the nested double try catch code here

Reply:
The inner try/catch is dealing with systems that lack dual stack support - we expect exceptions there, and can safely ignore them. The outer try/catch is for unexpected exceptions which can't be ignored.

Reply:
Can you add this explanation as a comment to the code. Someone, at sometime will have the same question, and a comment here would explain this.